core/src/main/java/org/apache/carbondata/core/cache/dictionary/DoubleArrayTrieDictionary.java

```diff
@@ -0,0 +1,416 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.carbondata.core.cache.dictionary;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+/**
+ * A dictionary based on DoubleArrayTrie data structure that maps enumerations
+ * of byte[] to int IDs. With DoubleArrayTrie the memory footprint of the mapping
+ * is minimize,d if compared to HashMap.
+ * This DAT implementation is based on https://linux.thai.net/~thep/datrie/datrie.html
```

Comment:
Suggest changing to : This DAT implementation is spired by https://linux.thai.net/~thep/datrie/datrie.html